wsmaster/codenvy-hosted-api-audit/src/main/java/com/codenvy/api/audit/server/AuditManager.java

```diff
@@ -0,0 +1,178 @@
+        } catch (Exception exception) {
+            if (auditReport != null) {
+                deleteReportDirectory(auditReport);
+            }
+            LOG.error(exception.getMessage(), exception);
+            throw new ServerException(exception.getMessage(), exception);
+        } finally {
+            inProgress.set(false);
+        }
+
+        return auditReport;
+    }
+
+    private void printAllUsers(Path auditReport) throws ServerException {
+        Page<UserImpl> currentPage = userManager.getAll(30, 0);
+        do {
+            //Print users with their workspaces from current page
+            for (UserImpl user : currentPage.getItems()) {
+                List<WorkspaceImpl> workspaces;
+                int permissionsNumber;
+                try {
+                    workspaces = workspaceManager.getWorkspaces(user.getId());
+                    permissionsNumber = workspaces.size();
+                    //add workspaces witch are belong to user, but user doesn't have permissions for them.
+                    workspaceManager.getByNamespace(user.getName())
+                                    .stream()
+                                    .filter(workspace -> !workspaces.contains(workspace))
+                                    .forEach(workspaces::add);
+                } catch (ServerException exception) {
+                    reportPrinter.printError("Failed to receive list of related workspaces for user " + user.getId(), auditReport);
+                    continue;
+                }
+                Map<String, List<AbstractPermissions>> wsPermissions = new HashMap<>();
+                for (WorkspaceImpl workspace : workspaces) {
+                    try {
+                        wsPermissions.put(workspace.getId(), permissionsManager.getByInstance("workspace", workspace.getId()));
+                    } catch (NotFoundException | ConflictException e) {
+                        wsPermissions.put(workspace.getId(), emptyList());
+                    }
+                }
+                reportPrinter.printUserInfoWithHisWorkspacesInfo(auditReport, user, workspaces, permissionsNumber, wsPermissions);
+            }
+            //Initialize next page if exist, otherwise stop printing report
+            final Optional<PageRef> nextPageRefOpt = currentPage.getNextPageRef();
+            if (nextPageRefOpt.isPresent()) {
+                final PageRef nextPageRef = nextPageRefOpt.get();
+                final long itemsBefore = nextPageRef.getItemsBefore();
+                //TODO: fix when https://github.com/eclipse/che/issues/2524 will be resolved
+                if (itemsBefore < Integer.MAX_VALUE) {
+                    throw new ServerException("Skip count limit was reached while retrieving all users");
+                }
+                currentPage = userManager.getAll(nextPageRef.getPageSize(), (int)itemsBefore);
+            } else {
+                break;
+            }
+        } while (true);
```

Comment:
I would move 
```
+            //Initialize next page if exist, otherwise stop printing report
 +            final Optional<PageRef> nextPageRefOpt = currentPage.getNextPageRef();
 +            if (nextPageRefOpt.isPresent()) {
 +                final PageRef nextPageRef = nextPageRefOpt.get();
 +                final long itemsBefore = nextPageRef.getItemsBefore();
 +                //TODO: fix when https://github.com/eclipse/che/issues/2524 will be resolved
 +                if (itemsBefore < Integer.MAX_VALUE) {
 +                    throw new ServerException("Skip count limit was reached while retrieving all users");
 +                }
 +                currentPage = userManager.getAll(nextPageRef.getPageSize(), (int)itemsBefore);
 +            } else {
 +                break;
 +            }
```
to private method and rewrite cycle as 
```
do {
...
} while ((currentPage = doGetNextPage) != null);
```